src/composer.rs

```diff
@@ -761,6 +765,57 @@ mod test {
         test_composer_with_pk_vk(constraint_system, vec![case_1]);
     }
 
+    #[test]
+    fn test_compute_merkle_root_constraint() {
+        use tempfile::tempdir;
+        let temp_dir = tempdir().unwrap();
```

Comment:
not blocking: perhaps we could find a better way to test this such that we can re-use infrastructure from the merkle tests